src/crystal/environment.cr

```diff
@@ -0,0 +1,35 @@
+module Crystal
+  # ```
+  # require "crystal/environment"
+  #
+  # Crystal.env.name         # => "development"
+  # Crystal.env.development? # => true
+  # Crystal.env.production?  # => false
+  # ```
+  module Environment
+    ENV_KEY     = "CRYSTAL_ENV"
+    ENV_VALUES  = %i(development test staging production)
```

Comment:
What if I want other environment names? What if I don't want `staging`, because I consider that staging == production?

Reply:
These values are used only to auto-generate query methods like `Crystal.env.development?`.
In 99% of cases these should be enough, but if you don't need/want to use those, you can always do `Crystal.env.name == "foo"` to check for non-standard env.

Reply:
This is not nice:

```crystal
if Crystal.env.staging? || Crystal.env.name == "integration"
  # do some stuff
end
```

Reply:
I agree, in terms of solutions we could add `Crystal::Environment.setup(keys)` macro replacing/extending defaults, or use `method_missing` (which was deemed a bad idea before).

Reply:
I like the idea for a setup macro. Though I think the default envs are going to be enough in most cases

Reply:
Also I think staging should be removed. Amber uses dev/test/prod and so does Elixir’s mix, Rails etc. Heroku even recommends against staging. So I think the main 3 should be there and people can add others if they want. Thoughts?

Reply:
PoC implemented, see https://github.com/Sija/crystal-environment/blob/master/spec/crystal-environment_spec.cr#L51